src/pages/Rockets.js

```diff
@@ -1,7 +1,50 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchRocketsFromServer, bookRocket, cancelBooking } from '../redux/rockets/rockets';
 
-const Rockets = () => (
-  <div>Rockets</div>
-);
+const Rockets = () => {
+  const rockets = useSelector((state) => state.rockets);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    if (!rockets.length) dispatch(fetchRocketsFromServer());
+  }, []);
+
+  return (
+    <div className="rocket">
+      {rockets.map((rocket) => {
+        const {
+          id, name, description, images, reserved,
+        } = rocket;
+
+        return (
+          <div key={id} className="eachRocket">
+            <img src={images} alt="rocket" className="rocket-image" />
+            <div className="rocket-info">
+              <h3>{name}</h3>
+              <p className="reserve-sect">
+                {reserved === true && (
+                <span className="rev-word">
+                  Reserved
+                </span>
+                )}
+                {' '}
+                {description}
+              </p>
+              {reserved ? (
+                <button type="button" className="cancel-book" onClick={() => dispatch(cancelBooking(id))}>
+                  Cancel Reservation
+                </button>
+              ) : (
+                <button className="reserveBtn" type="button" onClick={() => dispatch(bookRocket(id))}>Reserve Rocket</button>
```

Comment:
[optional] - I think you should consider using the reusable button component and pass in your desired props to ensure consistency in the code base. Having a resuable components helps in preventing duplicates 

import Button from'./Button'

`<Button type="button" className="cancel-book" onClick={() => dispatch(cancelBooking(id))}>
                  Cancel Reservation
                </Button>
              ) : (
                <Button className="reserveBtn" type="button" onClick={() => dispatch(bookRocket(id))}>Reserve Rocket</Button `

Reply:
OK I will consider it . Thanks for suggestion 👍